openhtf/exe/__init__.py

```diff
@@ -36,6 +36,11 @@ class TestStopError(Exception):
   """Test is being stopped."""
 
 
+def _InvalidTestStart():
```

Comment:
We assume it'll be set by the end of the test.  This was originally hacked into XTF because of board test - you don't know the USB serial until you power up the device, which required starting the test without knowing the serial yet.  This turned out to be a common enough case that it's been built into OpenHTF for a while now.  In fact, the ability to set the `dut_id` by returning it from the `test_start` callback is really a convenience at this point.